examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/ContentLauncherLaunchURLExampleFragment.java

```diff
@@ -0,0 +1,153 @@
+    this.selectedCastingPlayer = selectedCastingPlayer;
+  }
+
+  /**
+   * Use this factory method to create a new instance of this fragment using the provided
+   * parameters.
+   *
+   * @param selectedCastingPlayer CastingPlayer that the casting app connected to
+   * @return A new instance of fragment ContentLauncherLaunchURLExampleFragment.
+   */
+  public static ContentLauncherLaunchURLExampleFragment newInstance(
+      CastingPlayer selectedCastingPlayer) {
+    return new ContentLauncherLaunchURLExampleFragment(selectedCastingPlayer);
+  }
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+  }
+
+  @Override
+  public View onCreateView(
+      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    this.launchUrlButtonClickListener =
+        v -> {
+          List<Endpoint> endpoints = selectedCastingPlayer.getEndpoints();
+          if (endpoints == null) {
+            Log.e(TAG, "No Endpoints found on CastingPlayer");
+            return;
+          }
+
+          Endpoint endpoint =
+              endpoints
+                  .stream()
+                  .filter(e -> SAMPLE_ENDPOINT_VID.equals(e.getVendorId()))
+                  .findFirst()
+                  .get();
+          if (endpoint == null) {
+            Log.e(
+                TAG,
+                "No Endpoint with chosen vendorID: "
+                    + SAMPLE_ENDPOINT_VID
+                    + " found on CastingPlayer");
+            return;
+          }
+
+          if (!endpoint.hasCluster(MatterClusters.ContentLauncherCluster.class)) {
+            Log.e(TAG, "Endpoint with chosen vendorID does not support ContentLauncher cluster");
+            return;
+          }
+
+          MatterClusters.ContentLauncherCluster cluster =
+              endpoint.getCluster(MatterClusters.ContentLauncherCluster.class);
+          MatterCommands.ContentLauncherClusterLaunchURLCommand command =
+              cluster.getCommand(MatterCommands.ContentLauncherClusterLaunchURLCommand.class);
+          if (command == null) {
+            Log.e(
+                TAG,
+                "ContentLauncher cluster on Endpoint with chosen vendorID does not support LaunchURL command");
+            return;
+          }
+
+          // create the request object from GUI inputs
+          MatterCommands.ContentLauncherClusterLaunchURLRequest request =
+              new MatterCommands.ContentLauncherClusterLaunchURLRequest();
+          request.contentURL =
+              ((EditText) getView().findViewById(R.id.contentUrlEditText)).getText().toString();
+          request.displayString =
+              ((EditText) getView().findViewById(R.id.contentDisplayStringEditText))
+                  .getText()
+                  .toString();
+          CompletableFuture<MatterCommands.ContentLauncherClusterResponse> responseFuture =
+              command.invoke(request, 5000);
```

Comment:
I would suggest to use 
https://github.com/project-chip/connectedhomeip/blob/2f2c4f1e311c9c35912e21156f816837986bc857/src/controller/java/src/chip/devicecontroller/ChipDeviceController.java#L1192

AndroidInteractionClient.cpp and its JNI

https://github.com/project-chip/connectedhomeip/blob/2f2c4f1e311c9c35912e21156f816837986bc857/src/controller/java/AndroidInteractionClient.cpp

Reply:
We should try our best to leverage the existing Java/JNI code for InteractionModel and reduce the duplicate maintenance effort.